packages/algob/src/internal/deployer.ts

```diff
@@ -284,6 +348,98 @@ class DeployerBasicMode {
+   * Case 1: If it exist check if that info is deleted or not by checking deleted boolean
+   * If deleted boolean is true throw error
+   * else, pass
+   * Case 2: If it doesn't exist, pass
+   * @param asset asset index or asset name
+   */
+  private assertIfASAExist (asset: string | number): void {
+    let key, res;
+    if (typeof asset === "string") {
+      res = this.asa.get(asset);
+    } else if (typeof asset === "number") {
+      key = this.getAssetCheckpointKeyFromIndex(asset);
+      res = key ? this.asa.get(key) : undefined;
+    }
+    if (res?.deleted === true) {
+      throw new BuilderError(
+        ERRORS.GENERAL.ASSET_DELETED, {
+          asset: asset
+        });
+    }
+  }
+
+  /**
+   * Check if SSC info exist for app id,
+   * First: search for SSCInfo in checkpoints
+   * Case 1: If it exist check if that info is deleted or not by checking deleted boolean
+   * If deleted boolean is true throw error
+   * else, pass
+   * Case 2: If it doesn't exist, pass
+   * @param appID Application index
+   */
+  private assertIfAppExist (appID: number): void {
+    const key = this.getAppCheckpointKeyFromIndex(appID);
+    const res = key ? this.getSSCfromCPKey(key) : undefined;
+    if (res?.deleted === true) {
+      throw new BuilderError(
+        ERRORS.GENERAL.APP_DELETED, {
+          app: appID
+        });
+    }
+  }
+
+  /**
+   * Group transactions into asa and app, check for cp deletion
+   * @param txn Transaction execution parameter
+   */
+  private _internalDeletedCPCheck (txn: rtypes.ExecParams): void {
+    switch (txn.type) {
+      case rtypes.TransactionType.ModifyAsset:
+      case rtypes.TransactionType.FreezeAsset:
+      case rtypes.TransactionType.RevokeAsset:
+      case rtypes.TransactionType.OptInASA:
+      case rtypes.TransactionType.DestroyAsset: {
+        this.assertIfASAExist(txn.assetID);
+        break;
+      }
+      case rtypes.TransactionType.TransferAsset: {
+        // If transaction is not opt-out check for CP deletion
+        if (txn.payFlags.closeRemainderTo === undefined) {
+          this.assertIfASAExist(txn.assetID);
+        }
+        break;
+      }
+      case rtypes.TransactionType.DeleteSSC:
+      case rtypes.TransactionType.OptInSSC:
+      case rtypes.TransactionType.UpdateSSC:
+      case rtypes.TransactionType.CallNoOpSSC: {
+        this.assertIfAppExist(txn.appID);
+        break;
+      }
+    }
+  }
+
+  /**
+   * Checks if checkpoint is deleted for a particular transaction
+   * if checkpoint exist and is marked as deleted,
+   * throw error(except for opt-out transactions), else pass
+   * @param execParams Transaction execution parameters
+   */
+  checkForDeletedCP (execParams: rtypes.ExecParams | rtypes.ExecParams[]): void {
+    if (Array.isArray(execParams)) {
+      for (const txn of execParams) {
+        this._internalDeletedCPCheck(txn);
+      }
+    } else {
+      this._internalDeletedCPCheck(execParams);
```

Comment:
Why using internal prefix? I think `_assertCpNotDeleted` is clear

Reply:
not any specific reason, just out of names, this seemed easily understandable